gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMapper.java

```diff
@@ -188,21 +155,55 @@ public Application convert(
 
         application.setOwner(owner);
         application.setUpdatedAt(applicationListItem.getUpdatedAt().toInstant().atOffset(ZoneOffset.UTC));
-        ApplicationSettings settings = applicationListItem.getSettings();
-        application.setHasClientId(
-            settings != null &&
-            (
-                (
-                    settings.getOauth() != null && settings.getOauth().getClientId() != null && !settings.getOauth().getClientId().isEmpty()
-                ) ||
-                (settings.getApp() != null && settings.getApp().getClientId() != null && !settings.getApp().getClientId().isEmpty())
-            )
-        );
+
+        convertSettings(applicationListItem.getSettings(), application);
+
         if (applicationListItem.getApiKeyMode() != null) {
             application.setApiKeyMode(ApiKeyModeEnum.valueOf(applicationListItem.getApiKeyMode().name()));
         } else {
             application.setApiKeyMode(ApiKeyModeEnum.UNSPECIFIED);
         }
         return application;
     }
+
+    private static void convertSettings(ApplicationSettings applicationEntitySettings, Application application) {
+        if (applicationEntitySettings != null) {
+            io.gravitee.rest.api.portal.rest.model.ApplicationSettings appSettings =
+                new io.gravitee.rest.api.portal.rest.model.ApplicationSettings();
+
+            final SimpleApplicationSettings simpleAppEntitySettings = applicationEntitySettings.getApp();
+            if (simpleAppEntitySettings != null) {
+                appSettings.app(
+                    new io.gravitee.rest.api.portal.rest.model.SimpleApplicationSettings()
+                        .clientId(simpleAppEntitySettings.getClientId())
+                        .type(simpleAppEntitySettings.getType())
+                );
+                application.setHasClientId(simpleAppEntitySettings.getClientId() != null);
+            } else {
+                final OAuthClientSettings oAuthClientEntitySettings = applicationEntitySettings.getOauth();
+
+                if (Objects.nonNull(oAuthClientEntitySettings)) {
```

Comment:
We should be consistent.
Either you use "Objects.nonNull" everywhere, or you use "!= null".

Reply:
Done